paddle/fluid/memory/allocation/buffered_allocator.cc

```diff
@@ -50,29 +57,32 @@ void BufferedAllocator::FreeCache(size_t size) {
 bool BufferedAllocator::IsAllocThreadSafe() const {
   return this->underlying_allocator_->IsAllocThreadSafe();
 }
+
 void BufferedAllocator::Free(Allocation *allocation) {
   platform::LockGuardPtr<std::mutex> guard(mtx_);
   allocations_.emplace(allocation->size(), AllocationPtr(allocation));
+  cache_size_.fetch_add(allocation->size());
```

Comment:
why not just cache_size_ += allocation->size()?  since we do not need the previous held value